api/client/charms/localcharmclient.go

```diff
@@ -92,7 +98,15 @@ func (c *LocalCharmClient) AddLocalCharm(curl *charm.URL, ch charm.Charm, force
 		return nil, errors.Errorf("invalid charm %q: has no hooks nor dispatch file", curl.Name)
 	}
 
-	newCurlStr, err := c.charmPutter.PutCharm(context.Background(), curl.String(), archive)
+	hash, err := hashArchive(archive)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	charmRef := fmt.Sprintf("%s-%s", curl.Name, hash)
+
+	modelTag, _ := c.facade.RawAPICaller().ModelTag()
```

Comment:
Why are we dropping an error?

Reply:
This isn't an error, it's a boolean which returns if this is a 'controller-only connection'.

It's value isn't required